test/test/coro_collections.cpp

```diff
@@ -0,0 +1,38 @@
+// Intentionally not using pch...
```

Comment:
I'd recommend adding a test for GetMany(), as well as normally iterating e.g. `for (hstring h : Generator())`).

What about value types and reference types?

Reply:
I'm curious what's the behavior regarding AddRef/Release counts. Also, it might be a good idea to add a test checking that not completing the iteration but dropping the IIterable & IIterator will free the coroutine frame.